Failure to import Google annotations proto

<!--
### Are you looking for help?

If you have a [Lightbend Subscription](https://www.lightbend.com/lightbend-platform-subscription), please reach out via the [Lightbend Portal](https://portal.lightbend.com/).

This is an issue tracker used to manage and track the development of this particular module.

Please report issues regarding other projects in their respective issue trackers, e.g.:
 - Akka:          https://github.com/akka/akka/issues 
 - Akka HTTP:     https://github.com/akka/akka-http/issues 
 - Alpakka:       https://github.com/akka/alpakka/issues 
 - Alpakka Kafka: https://github.com/akka/alpakka-kafka/issues 

Please ask questions or discuss ideas in the [discuss.akka.io forum](https://discuss.akka.io/).

## Please add the following sections to your bug report
-->

### Versions used 

Akka version: 1.1.1

### Expected Behavior

When creating the Java files from protobuf in Maven with
```
            <plugin>
                <groupId>com.lightbend.akka.grpc</groupId>
                <artifactId>akka-grpc-maven-plugin</artifactId>
                <version>${akka.grpc.version}</version>
                <configuration>
                    <language>Java</language>
                    <includeStdTypes>true</includeStdTypes>
                    <protoPaths>
                        <protoPath>my_protos_path</protoPath>
                    </protoPaths>
                </configuration>
                <executions>
                    <execution>
                        <goals>
                            <goal>generate</goal>
                        </goals>
                    </execution>
                </executions>
            </plugin>
```
I'd expect the compiler to be able to pick up the standard file `google/api/annotations.proto` and correctly compile the protobufs.

### Actual Behavior

The compilation fails with
```
[ERROR] Failed to execute goal com.lightbend.akka.grpc:akka-grpc-maven-plugin:1.1.1:generate (default) on project registry: Execution default of goal com.lightbend.akka.grpc:akka-grpc-maven-plugin:1.1.1:generate failed: protoc exit code 1: google/api/annotations.proto: File not found.
```

For comparison, using
```
            <plugin>
                <groupId>org.xolstice.maven.plugins</groupId>
                <artifactId>protobuf-maven-plugin</artifactId>
                <version>0.6.1</version>
                <configuration>
                    <protocArtifact>com.google.protobuf:protoc:3.8.0:exe:${os.detected.classifier}</protocArtifact>
                    <pluginId>grpc-java</pluginId>
                    <pluginArtifact>io.grpc:protoc-gen-grpc-java:1.16.1:exe:${os.detected.classifier}</pluginArtifact>
                    <protoSourceRoot>my_protos_path</protoSourceRoot>
                </configuration>
                <executions>
                    <execution>
                        <goals>
                            <goal>compile</goal>
                            <goal>compile-custom</goal>
                        </goals>
                    </execution>
                </executions>
            </plugin>
```
correctly creates the Java implementations.

I have also tried configuring `<protocVersion>-v3.8.0</protocVersion>` to match the versions but it gives the same error.


### Relevant logs

Relevant build logs shared above.

### Reproducible Test Case

I'm currently unsure what's the best way to provide the reproducibility instructions. [This](https://github.com/grpc-ecosystem/grpc-gateway/blob/master/examples/internal/helloworld/helloworld.proto) example could be used as a base proto.

--- a/docs/src/main/paradox/buildtools/maven.md
+++ b/docs/src/main/paradox/buildtools/maven.md
@@ -77,10 +77,53 @@ which is a relative path to the project basedir. The below configuration overrid
 In gRPC it is common to make the version of the protocol you are supporting
 explicit by duplicating the proto definitions in your project.
 
-When using @ref[sbt](sbt.md) as a build system, we also support loading your
-proto definitions from a dependency classpath. This is not yet supported
-for Maven and @ref[Gradle](gradle.md). If you are interested in this feature
-it is tracked in issue [#152](https://github.com/akka/akka-grpc/issues/152).
+When using @ref[sbt](sbt.md) or @ref[Gradle](gradle.md) as a build system, we also support loading your
+proto definitions from a dependency classpath.
+
+For Maven, you can achieve a similar result by using the `maven-dependency-plugin` to unpack the
+proto files from a dependency and then adding the unpacked directory as a `protoPath`:
+
+`pom.xml`
+:   ```xml
+    <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-dependency-plugin</artifactId>
+        <executions>
+            <execution>
+                <id>unpack-proto</id>
+                <phase>generate-sources</phase>
+                <goals>
+                    <goal>unpack</goal>
+                </goals>
+                <configuration>
+                    <artifactItems>
+                        <artifactItem>
+                            <groupId>com.example</groupId>
+                            <artifactId>my-protos</artifactId>
+                            <version>1.0.0</version>
+                            <type>jar</type>
+                            <outputDirectory>${project.build.directory}/protos-dep</outputDirectory>
+                        </artifactItem>
+                    </artifactItems>
+                </configuration>
+            </execution>
+        </executions>
+    </plugin>
+    <plugin>
+        <groupId>org.apache.pekko</groupId>
+        <artifactId>pekko-grpc-maven-plugin</artifactId>
+        <version>${pekko.grpc.version}</version>
+        <configuration>
+            <protoPaths>
+                <protoPath>src/main/protobuf</protoPath>
+                <protoPath>${project.build.directory}/protos-dep</protoPath>
+            </protoPaths>
+        </configuration>
+    </plugin>
+    ```
+
+All configured `protoPaths` are passed as `-I` import parameters to `protoc`, allowing proto files
+in your project to import definitions from the unpacked artifact.
 
 ## JDK 8 support
 

--- a/maven-plugin/src/main/scala/org/apache/pekko/grpc/maven/AbstractGenerateMojo.scala
+++ b/maven-plugin/src/main/scala/org/apache/pekko/grpc/maven/AbstractGenerateMojo.scala
@@ -129,21 +129,27 @@ abstract class AbstractGenerateMojo @Inject() (buildContext: BuildContext) exten
 
   def addGeneratedSourceRoot(generatedSourcesDir: String): Unit
 
+  // https://maven.apache.org/plugin-developers/common-bugs.html#Resolving_Relative_Paths
+  def normalize(protoPath: String): File = {
+    val protoFile = new File(protoPath)
+    if (!protoFile.isAbsolute()) {
+      new File(project.getBasedir(), protoPath).toPath().normalize().toFile()
+    } else {
+      protoFile
+    }
+  }
+
+  lazy val normalizedProtoPaths: Seq[File] = {
+    import scala.jdk.CollectionConverters._
+    protoPaths.asScala.map(normalize).toSeq
+  }
+
   override def execute(): Unit = {
     val chosenLanguage = parseLanguage(language)
 
     var directoryFound = false
-    protoPaths.forEach { protoPath =>
-      // verify proto dir exists
-      // https://maven.apache.org/plugin-developers/common-bugs.html#Resolving_Relative_Paths
-      val protoDir = {
-        val protoFile = new File(protoPath)
-        if (!protoFile.isAbsolute()) {
-          new File(project.getBasedir(), protoPath).toPath().normalize().toFile()
-        } else {
-          protoFile
-        }
-      }
+
+    normalizedProtoPaths.foreach { protoDir =>
       if (protoDir.exists()) {
         directoryFound = true
         // generated sources should be compiled
@@ -218,15 +224,16 @@ abstract class AbstractGenerateMojo @Inject() (buildContext: BuildContext) exten
   private[this] def executeProtoc(
       protocCommand: Seq[String] => Int,
       schemas: Set[File],
-      protoDir: File,
       protocOptions: Seq[String],
       targets: Seq[Target]): Int =
     try {
-      val incPath = "-I" + protoDir.getCanonicalPath
+      val protocIncludePaths = normalizedProtoPaths.map { includePath =>
+        "-I" + includePath.getCanonicalPath
+      }
       protocbridge.ProtocBridge.execute(
         ProtocRunner.fromFunction((args, _) => protocCommand(args)),
         targets,
-        Seq(incPath) ++ protocOptions ++ schemas.map(_.getCanonicalPath),
+        protocIncludePaths ++ protocOptions ++ schemas.map(_.getCanonicalPath),
         artifact =>
           throw new RuntimeException(
             s"The version of sbt-protoc you are using is incompatible with '${artifact}' code generator. Please update sbt-protoc to a version >= 0.99.33"))
@@ -265,7 +272,7 @@ abstract class AbstractGenerateMojo @Inject() (buildContext: BuildContext) exten
 
       getLog.info("Compiling protobuf")
       val (out, err, exitCode) = captureStdOutAndErr {
-        executeProtoc(protocCommand, schemas, protoDir, protocOptions, generatedTargets)
+        executeProtoc(protocCommand, schemas, protocOptions, generatedTargets)
       }
       if (exitCode != 0) {
         err.split("\n\r").map(_.trim).map(parseError).foreach {

